### PR TITLE
Don't silently ignore bad global config files.

### LINKED ISSUE #1554: cylc global config: bad configs are silently ignored
If I add something invalid like this to my `~/.cylc/global.rc`:
```
[pyro]
    base port = wellington
```
then my whole user config is silently ignored. The same happens for site configs.

It should probably just fall over with a 'bad user' error.

### PR DESCRIPTION
**Pull request by hjoliver**
Close #1554.

For global and GUI config files,
 * warn about bad site files (which users can't fix)
 * abort on bad user files (which they can fix)

@benfitzpatrick - please review 1.

### CODE REVIEW

**Comment by hjoliver** [comment]
My test battery passes.

**Comment by benfitzpatrick** [comment]
Manually tests OK

**Comment by hjoliver** [comment]
Branch updated.

**Comment by benfitzpatrick** [comment]
@matthewrmshin, please review 2.

File: lib/parsec/fileparse.py [hunk 1]
```diff
@@ -153,7 +154,7 @@ def addict( cfig, key, val, parents, index ):
     if not isinstance( cfig, dict ):
         # an item of this name has already been encountered at this level
         print >> sys.stderr, itemstr( parents, key, val )
-        raise ParseError( 'ERROR line ' + str(index) + ': already encountered ' + itemstr( parents ))
+        raise FileParseError( 'ERROR line ' + str(index) + ': already encountered ' + itemstr( parents ))
```
Comment (hjoliver) [inline lib/parsec/fileparse.py]: "ParseError" looked too much like the general "ParsecError".

**Comment by matthewrmshin** [comment]
Unfortunately, merging #1500 means that this branch needs an update.

**Comment by hjoliver** [comment]
Fortunately, the conflicts were pretty trivial.

**Pull request merged**